dask/dataframe/core.py

```diff
@@ -562,6 +563,41 @@ def std(self, axis=None, ddof=1):
             name = '{0}std(ddof={1})'.format(self._token_prefix, ddof)
             return map_partitions(np.sqrt, no_default, v, token=name)
 
+    def quantile(self, q=0.5, axis=0):
+        """ Approximate row-wise and precise column-wise quantiles of DataFrame
+
+        q : list/array of floats, default 0.5 (50%)
+            Iterable of numbers ranging from 0 to 1 for the desired quantiles
+        axis : {0, 1, 'index', 'columns'} (default 0)
+            0 or 'index' for row-wise, 1 or 'columns' for column-wis
+        """
+        axis = self._validate_axis(axis)
+        name = 'quantiles-concat--' + tokenize(self, q, axis)
+
+        if axis == 1:
+            if isinstance(q, list):
+                # Not supported, the result will have current index as columns
+                raise ValueError("'q' must be scalar when axis=1 is specified")
+            return map_partitions(pd.DataFrame.quantile, None, self,
+                                  q, axis, token=name)
+        else:
+            num = self._get_numeric_data()
+            quantiles = tuple(quantile(self[c], q) for c in num.columns)
+
+            dask = {}
+            dask = merge(dask, *[q.dask for q in quantiles])
+            qnames = [(q._name, 0) for q in quantiles]
+
+            if isinstance(quantiles[0], Scalar):
+                dask[(name, 0)] = (pd.Series, (list, qnames), num.columns)
+                divisions = (min(num.columns), max(num.columns))
```

Comment:
It would be a bit of work, but we could also add `dtypes` to the metadata that we track.